bin/sidecli.re

```diff
@@ -749,8 +760,123 @@ let remove_trusted_validator = {
+  Term.info(
+    "propose-new-validator",
+    ~version="%‌%VERSION%%",
+    ~doc,
+    ~exits,
+    ~man,
+  );
+};
+
+let propose_new_validator = (folder_node, validator_address) => {
+  open Networking;
+  let.await identity = read_identity(~node_folder=folder_node);
+  let.await block_level_response = request_block_level((), identity.uri);
+  let block_level = block_level_response.level;
+  let membership_change =
+    Operation.Side_chain.sign(
+      ~secret=identity.key,
+      ~nonce=0l,
+      ~block_height=block_level,
+      ~source=Wallet.of_address(identity.t),
+      ~kind=Add_validator({address: validator_address}),
+    );
+
+  let.await () =
+    Networking.request_operation_gossip(
+      Networking.Operation_gossip.{operation: membership_change},
+      identity.uri,
+    );
+  Format.printf(
+    "operation.hash: %s\n%!",
+    BLAKE2B.to_string(membership_change.hash),
+  );
+
+  Lwt.return(`Ok());
+};
+
+let propose_new_validator = {
+  let validator_address = {
+    let doc = "Address of the proposed validator being added to the network";
+    let env = Arg.env_var("NEW_VALIDATOR_ADDRESS", ~doc);
+    Arg.(
+      required
+      & pos(2, some(address), None)
+      & info([], ~env, ~docv="new_validator_address", ~doc)
+    );
+  };
+
+  Term.(
+    lwt_ret(const(propose_new_validator) $ folder_node $ validator_address)
+  );
+};
+
+let info_propose_validator_removal = {
+  let doc = "Propose removing a validator from the network.";
+  Term.info(
+    "propose-validator-removal",
+    ~version="%‌%VERSION%%",
+    ~doc,
+    ~exits,
+    ~man,
+  );
+};
+
+let propose_validator_removal =
+    (folder_node, validator_address) => {
+  open Networking;
+  let.await identity = read_identity(~node_folder=folder_node);
+  let.await block_level_response = request_block_level((), identity.uri);
+  let block_level = block_level_response.level;
+  let membership_change =
+    Operation.Side_chain.sign(
+      ~secret=identity.key,
+      ~nonce=0l,
+      ~block_height=block_level,
+      ~source=Wallet.of_address(identity.t),
+      ~kind=Remove_validator({address: validator_address}),
+    );
+
+  let.await () =
+    Networking.request_operation_gossip(
+      Networking.Operation_gossip.{operation: membership_change},
+      identity.uri,
+    );
+  Format.printf(
+    "operation.hash: %s\n%!",
+    BLAKE2B.to_string(membership_change.hash),
+  );
+
+  Lwt.return(`Ok());
+};
+
+let propose_validator_removal = {
+  let validator_address = {
+    let doc = "Address of the proposed validator being removed from the network";
+    let env = Arg.env_var("REMOVED_VALIDATOR_ADDRESS", ~doc);
+    Arg.(
+      required
+      & pos(2, some(address), None)
+      & info([], ~env, ~docv="removed_validator_address", ~doc)
+    );
+  };
+
+  Term.(
+    lwt_ret(
+      const(propose_validator_removal)
+      $ folder_node
+      $ validator_address,
+    )
+  );
+};
+
+// Run the CLI
+Mirage_crypto_rng_unix.initialize();
```

Comment:
This is not needed anymore

Reply:
RNG isn't being properly initialised. I discovered this while testing.

Reply:
Yes, fixed that on main